### PR TITLE
Add plugin section to release docs

### PR DESCRIPTION
**Pull request by dsimansk**
First pass on the release docs update. 

Just going trough and documenting the current steps, it feels like half of that should be already scripted. :see_no_evil: 

/cc @rhuss @lance @matzew

### CODE REVIEW

**Review by lance — COMMENTED** [review]
Some minor wording suggestions and tweaks. Otherwise lgtm

File: RELEASE.md [hunk 1]
```diff
@@ -132,6 +132,109 @@ git push
 ```
 Create a new pull request to update `main` branch.
 
+### Plugins
+
+Currently, we ship 3 plugin within productized `kn` binary. It's achieved by mechanism called "plugin inlining", for futher reference see upstream's docs.
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply RELEASE.md:137]: Sure, I was a bit lazy here.

File: RELEASE.md [hunk 2]
```diff
@@ -132,6 +132,109 @@ git push
...
+  - name: kn-plugin-func
+    module: knative.dev/kn-plugin-func
+    pluginImportPath: knative.dev/kn-plugin-func
+    version: v0.23.1
+  - name: kn-plugin-event
+    module: knative.dev/kn-plugin-event
+    pluginImportPath: knative.dev/kn-plugin-event/pkg/plugin
+    version: v0.28.0
+```
+NOTE: knb is lacking midstream replace support. That's solved in further steps.
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply RELEASE.md:171]: It's actually a todo note highlighting a missing feature in `knb`. I'll remove it and rather fix `knb`. :)

**Comment by rhuss** [comment]
Looks good to me. @dsimansk do you still want to address the referenced issue (cleaning up references to future features, adding URL) ? Setting the pr to hold for now, feel free to unhold.

/hold
/lgtm
/approve

**Pull request closed without merging**